src/components/AlphabetFilter/AlphabetFilter.tsx

```diff
@@ -0,0 +1,197 @@
+import { Button } from "../Button/Button";
+import { Heading } from "../Heading/Heading";
+import { Text } from "../Text/Text";
+import useNYPLBreakpoints from "../../hooks/useNYPLBreakpoints";
+
+export interface AlphabetFilterProps {
+  /** Array of letters to specify which `Button` components should be set in an `enabled`
+   * state. By default, all buttons are `enabled`. */
+  activeLetters?: string[];
+  /** A class name for the AlphabetFilter parent div. */
+  className?: string;
+  /** The currentLetter can be used to programatically set the selected letter without the
+   * user explicitly requesting it. */
+  currentLetter?: string;
+  /** Value used to set the text for a `Text` component that will be displayed above the letter buttons. */
+  descriptionText?: string | JSX.Element;
+  /** Value used to set the text for a `Heading` component. */
+  headingText?: string;
+  /** ID that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** Adds the `disabled` prop to the AlphabetFilter when true. */
+  isDisabled?: boolean;
+  /** The callback function called when a letter button or the Show All button is clicked. */
+  onClick: (filterValue: string) => void;
+}
+
+export const AlphabetFilter = chakra(
+  forwardRef<HTMLDivElement, AlphabetFilterProps>((props, ref?) => {
+    const {
+      activeLetters,
+      className,
+      currentLetter,
+      descriptionText,
+      headingText,
+      id,
+      isDisabled,
+      onClick,
+      ...rest
+    } = props;
+
+    const styles = useMultiStyleConfig("AlphabetFilter", {});
+    const { isLargerThanMobile } = useNYPLBreakpoints();
+
+    const filterButtons = [
+      { text: "#", value: "#" },
+      { text: "A", value: "a" },
+      { text: "B", value: "b" },
+      { text: "C", value: "c" },
+      { text: "D", value: "d" },
+      { text: "E", value: "e" },
+      { text: "F", value: "f" },
+      { text: "G", value: "g" },
+      { text: "H", value: "h" },
+      { text: "I", value: "i" },
+      { text: "J", value: "j" },
+      { text: "K", value: "k" },
+      { text: "L", value: "l" },
+      { text: "M", value: "m" },
+      { text: "N", value: "n" },
+      { text: "O", value: "o" },
+      { text: "P", value: "p" },
+      { text: "Q", value: "q" },
+      { text: "R", value: "r" },
+      { text: "S", value: "s" },
+      { text: "T", value: "t" },
+      { text: "U", value: "u" },
+      { text: "V", value: "v" },
+      { text: "W", value: "w" },
+      { text: "X", value: "x" },
+      { text: "Y", value: "y" },
+      { text: "Z", value: "z" },
+      { text: "Show All", value: "showAll" },
+    ];
+
+    const refCurrentLetter = useRef(currentLetter);
+    const [selectedLetter, setSelectedLetter] = useState<string>(currentLetter);
+
+    // If the parent passes down a new currentLetter, and an onClick
+    // function exists, then set the internal state – selectedLetter –
```

Comment:
`onClick` is a required prop so when might it not be defined?

Reply:
Removed onClick statement since it will always be defined.